client/src/components/Editor/DraftEditor.js

```diff
@@ -0,0 +1,236 @@
+    EditorState.createEmpty(decorator)
+  );
+  //Amending code blocks for Prism
+  const getBlockStyle = (block) => {
+    switch (block.getType()) {
+      case "code-block":
+        return "language-".concat(mappedLanguage);
+      default:
+        return null;
+    }
+  };
+
+  // Rerender Draftjs every time we change language
+  useEffect(() => {
+    const selection = editorState.getSelection();
+    const block = editorState
+      .getCurrentContent()
+      .getBlockForKey(selection.getStartKey());
+    const data = block.getData().merge({ syntax: mappedLanguage });
+    const newBlock = block.merge({ data });
+    const newContentState = editorState.getCurrentContent().merge({
+      blockMap: editorState
+        .getCurrentContent()
+        .getBlockMap()
+        .set(selection.getStartKey(), newBlock),
+      selectionAfter: selection,
+    });
+    setEditorState(
+      EditorState.push(editorState, newContentState, "change-block-data")
+    );
+  }, [editorState, mappedLanguage]);
+
+  // Focus editor (we will chat using the same editor, right?)
+  const editor = useRef(null);
+  const focusEditor = () => {
+    editor.current.focus();
+  };
+  useEffect(() => {
+    focusEditor();
+  }, []);
+
+  //Change handler
+  const onChange = (newEditorState) => {
+    if (newEditorState.getCurrentContent().hasText()) {
+      hasContent(true);
+    } else {
+      hasContent(false);
+    }
+    setEditorState(EditorState.set(newEditorState, { decorator }));
+  };
+
+  // DraftJs code utils for easier code editing:
+  const handleKeyCommand = (command) => {
+    let newState;
+
+    if (CodeUtils.hasSelectionInBlock(editorState)) {
+      newState = CodeUtils.handleKeyCommand(editorState, command);
+    }
+
+    if (!newState) {
+      newState = RichUtils.handleKeyCommand(editorState, command);
+    }
+
+    if (newState) {
+      setEditorState(newState);
+      return "handled";
+    }
+    return "not-handled";
+  };
+
+  const keyBindingFn = (evt) => {
+    if (!CodeUtils.hasSelectionInBlock(editorState))
+      return Draft.getDefaultKeyBinding(evt);
+
+    const command = CodeUtils.getKeyBinding(evt);
+
+    return command || Draft.getDefaultKeyBinding(evt);
+  };
+
+  const handleReturn = (evt) => {
+    if (!CodeUtils.hasSelectionInBlock(editorState)) return "not-handled";
+
+    setEditorState(CodeUtils.handleReturn(evt, editorState));
+    return "handled";
+  };
+
+  const onTab = (evt) => {
+    if (!CodeUtils.hasSelectionInBlock(editorState)) return "not-handled";
+
+    setEditorState(CodeUtils.onTab(evt, editorState));
+    return "handled";
+  };
+
+  //Transfer text to make request
+  useEffect(() => {
+    if (makeSubmit) {
+      const rawText = convertToRaw(editorState.getCurrentContent());
+
+      onSubmit({ text: rawText });
+      setEditorState(EditorState.createEmpty(decorator));
+    }
+  }, [makeSubmit]);
+
+  // Rendering control buttons
+  const ToolButton = ({ format, icon }) => {
```

Comment:
Can we move these components outside of this component? Then we don't need to keep creating this new component every time inside this component and we may reuse this component